tests/drivers/uart/uart_elementary/src/main.c

```diff
@@ -0,0 +1,329 @@
+/*
+ * Copyright (c) 2024 Nordic Semiconductor ASA
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+/**
+ * @addtogroup t_uart_driver_uart_elementary
+ * @{
+ * @defgroup t_uart_elementary test_uart_elementary
+ * @brief TestPurpose: Verify elementary UART API methods that are not tested in other suites
+ * @}
+ */
+
+#include <zephyr/drivers/uart.h>
+#include <zephyr/ztest.h>
+
+#if DT_NODE_EXISTS(DT_NODELABEL(dut))
+#define UART_NODE DT_NODELABEL(dut)
+#else
+#define UART_NODE DT_CHOSEN(zephyr_console)
+#endif
+
+#if DT_NODE_EXISTS(DT_NODELABEL(dut_aux))
+#define UART_NODE_AUX DT_NODELABEL(dut_aux)
+#else
+#define UART_NODE_AUX DT_CHOSEN(zephyr_console)
+#endif
+
+#define SLEEP_TIME_US 1000
+#define TEST_BUFFER_LEN 10
+
+static const struct device *const uart_dev = DEVICE_DT_GET(UART_NODE);
+
+const uint8_t test_pattern[TEST_BUFFER_LEN] = { 0x11, 0x12, 0x13, 0x14, 0x15,
+						0x16, 0x17, 0x18, 0x19, 0x20 };
+uint8_t test_buffer[TEST_BUFFER_LEN] = {
+	0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00
```

Comment:
static variables are by default zeroed so it's redundant. same for `test_buffer_aux`.